mapillary_tools/sample_video.py

```diff
@@ -145,14 +145,24 @@ def sample_video(
 
 
 @contextmanager
-def wip_dir_context(wip_dir: Path, done_dir: Path):
+def wip_dir_context(wip_dir: Path, done_dir: Path, rename_timeout_sec:int=10):
     assert wip_dir != done_dir, "should not be the same dir"
     shutil.rmtree(wip_dir, ignore_errors=True)
     os.makedirs(wip_dir)
     try:
         yield wip_dir
         shutil.rmtree(done_dir, ignore_errors=True)
-        wip_dir.rename(done_dir)
+        error = None
+        renamed = False
+        start_time = time.time()
+        while (not renamed and time.time()-start_time < rename_timeout_sec):
+            try:
+                wip_dir.rename(done_dir)
+                renamed = True
+            except Exception as e:
```

Comment:
Sleep for 1 second here, or perhaps retry backoff?

Reply:
Without introducing new variables, it can be simplified to:
```python
while time.time() - start_time < rename_timeout_sec:
    try:
        wip_dir.rename(done_dir)
    except Exception as ex:
        sleep(1)
        continue
    else:
        break
```

Reply:
This fails to raise an exception in the event there is a legitimate error other than the PermissionErrors that are thrown by the os. This would lead to the wip directory being deleted by the finally clause without any alert being given to the user about the failure and may cause confusion. While this does simplify the code it could cause the debugging to be more confusing. 

I would argue for the code to remain as is, but since I am a guest in this repository I am happy to settle on this :)